build_tools/cmake/iree_run_module_test.cmake

```diff
@@ -0,0 +1,335 @@
+# Copyright 2022 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+# Utility function to return the platform name in crosscompile.
+function(iree_get_platform PLATFORM)
+  if(ANDROID AND CMAKE_ANDROID_ARCH_ABI STREQUAL "arm64-v8a")
+    set(_PLATFORM "android-arm64-v8a")
+  elseif(CMAKE_SYSTEM_PROCESSOR STREQUAL "x86_64")
+    set(_PLATFORM "x86_64")
+  else()
+    set(_PLATFORM "${CMAKE_SYSTEM_PROCESSOR}-${CMAKE_SYSTEM_NAME}")
+  endif()
+  set(${PLATFORM} "${_PLATFORM}" PARENT_SCOPE)
+endfunction()
+
+# iree_run_module_test()
+#
+# Creates a test using iree-run-module to run an IREE module (vmfb).
+#
+# The function is unimplemented in Bazel because it is not used there.
+#
+# Parameters:
+#   NAME: Name of the target
+#   MODULE_SRC: IREE module (vmfb) file.
+#   DRIVER: Driver to run the module with.
+#   RUNNER_ARGS: additional args to pass to iree-run-module. The driver
+#       and input file are passed automatically.
+#   EXPECTED_OUTPUT: A string representing the expected output from executing
+#       the module in the format accepted by `iree-run-module` or a file
+#       containing the same.
+#   LABELS: Additional labels to apply to the test. The package path and
+#       "driver=${DRIVER}" are added automatically.
+#   XFAIL: List of platforms (all, x86_64, android-arm64-v8a, riscv64-Linux,
```

Comment:
Maybe name this "XFAIL_PLATFORMS" to match "UNSUPPORTED_PLATFORMS"?

Reply:
Done.